src/mandelbrot.rs

```diff
@@ -0,0 +1,68 @@
+// The Computer Language Benchmarks Game
+// http://benchmarksgame.alioth.debian.org/
+//
+// contributed by the Rust Project Developers
+// contributed by TeXitoi
+// contributed by Matt Watson
+use std::io::Write;
+use std::io;
+use std::thread;
+const THREADS: usize = 8;
+const MAX_ITER: usize = 50;
+const DX: f64 = -1.5;
+const DY: f64 = -1.0;
+pub fn mbrotpt(x: f64, y: f64) -> usize {
+    let mut z = (0.0, 0.0);
+    for _ in 0..MAX_ITER {
+        z = (z.0 * z.0 - z.1 * z.1 + x,
+             2.0 * z.0 * z.1 + y);
+        if z.0 * z.0 + z.1 * z.1 >= 4.0 {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+fn mbrot8(x: usize, y: usize, inv: f64) -> u8 {
+    let mut result = 0 as usize;
```

Comment:
I prefer minimize type conversions: here, you can use directly u8, and it will be automatically inferred. You also must change the return type of mbrotpt to u8 for this to work.